### PR TITLE
Update PR Template

### PR DESCRIPTION
**Pull request by sovdeeth**
### Problem

The existing PR template was adequate, but didn't encourage very expressive PR descriptions. A single 'Description' header meant that users were basically faced with no guidance on how to write a PR description. This lead to many different forms of PR descriptions, some extremely detailed and some with barely any information. In addition, there was no clear link as to what issues this PR should close and which were related. MC and required plugin fields were hardly ever used and served no real purpose.

### Solution

This PR reworks the template so that there are clear prompts for the user to fill out and ensures the author spends time explaining the reasoning behind making the PR in the first place, as well as what the PR actually does. A 'Testing Completed' section has been added to help remind authors to include unit tests or test scripts, or at minimum describe the manual testing they did for the PR.

The MC version and Requirements fields have been removed as authors often forgot to fill them in and reviewers hardly ever referenced them. The Related field has been joined by a Completes field in order to clarify which issues should be marked as PR Available/Completed and which are just supplementary information or only partially completed by this PR.

This PR serves as an example of the template, showing the 4 main headers and the new 'Completes' field at the bottom.

### Testing Completed

This is a meta change to the GitHub repository and therefore has no behavior to test.

### Supporting Information

The Completes field also helps support further automation by giving a clear link between issues and their prs for a bot to follow. Following this PR, it will be possible to automatically label issues as PR available and Completed depending on the state of their PRs.

This PR also avoids the 'Closing keywords' format like `Closes #XXX` to avoid GitHub auto-closing issues as soon as the PR is merged, rather than when the PR is released.

---
**Completes:** none 
**Related:** none

### CODE REVIEW

**Review by Efnilite — APPROVED** [review]

**Review by erenkarakal — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,7 +1,19 @@
-### Description
-<!--- Describe your changes here. --->
+### Problem
+<!--- Why is this PR necessary? What problems exist that needed solving?  --->
+
+
+### Solution
+<!--- How your solution fixes the problem and a summary of the major code changes.  --->
```
Comment (APickledWalrus) [inline .github/pull_request_template.md:6]:
```suggestion
<!--- Explain how your solution fixes the problem and summarize the major code changes.  --->
```
My thought is that we might want to use more direct verbs

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+
+
+### Testing Completed
+<!--- List test scripts/unit tests and any manual testing that was performed. If no test scripts or unit test are present, explain why.  --->
```
Comment (APickledWalrus) [inline .github/pull_request_template.md:10]:
```suggestion
<!--- List test scripts/unit tests and any manual testing that was performed. If no test scripts or unit tests are present, explain why.  --->
```

**Review by sovdeeth — COMMENTED** [review]
test

**Review by UnderscoreTud — APPROVED** [review]
